### PR TITLE
homework_5

### PR DESCRIPTION
**Pull request by ArevikArakelyan**
please, my homework 5

### CODE REVIEW

File: src/algorithms/MathUtil.java [hunk 1]
```diff
@@ -37,10 +41,20 @@ public static int pow(int number, int pow){
...
+
     public static int factorial(int n) {
-       return 0;
+        int result = 1;
+
+        for (int i = 2; i <= n; i++) {
+            result *= i;
+        }
+
+        return result;
```
Comment (Karlen999) [inline src/algorithms/MathUtil.java:52]: done.

You can also write a recursive method

File: src/algorithms/MathUtil.java [hunk 2]
```diff
@@ -3,8 +3,10 @@
 public class MathUtil {
 
     public static void main(String[] args) {
-        System.out.println(pow(2, 0));
-
+        System.out.println(pow(3, 2));
+        System.out.println(factorial(5));
+        System.out.println(Math.abs(-5));
```
Comment (Karlen999) [inline src/algorithms/MathUtil.java:8]: You should write a method, not use a ready-made method Math.abs()

File: src/algorithms/MathUtil.java [hunk 3]
```diff
@@ -60,10 +74,14 @@ public static int abs(int n) {
...
+        int reversenum =0;
+        while( number != 0 )
+        {
+            reversenum = reversenum * 10;
+            reversenum = reversenum + number%10;
+            number = number/10;
+        }
 
-        return 0;
+        return reversenum;
```
Comment (Karlen999) [inline src/algorithms/MathUtil.java:85]: done

File: src/algorithms/MathUtil.java (hunk 1 above)
Comment (ArevikArakelyan) [reply src/algorithms/MathUtil.java:52]: public static int factorial(int n){
        int fact;
        if (n == 0)
            return 1;
        else {
            fact = n * factorial(n-1) ;
            return fact;
        }
    }    
I try, but something do not work

Comment (Karlen999) [reply src/algorithms/MathUtil.java:52]: it works correctly, you can optimize the code like this

if (n == 0) {
            return 1;
        }
        else
            return(n * factorial(n-1));

**Comment by Karlen999** [comment]
reviewed

**Pull request closed without merging**